python/lsst/meas/mosaic/mosaicTask.py

```diff
@@ -392,19 +393,19 @@ def readSrc(self, dataRef):
                     newMatch = table.makeRecord()
                     newMatch.assign(match[0], mapper)
                     match[0] = newMatch
-                primaryFluxKey = refSchema.find(refSchema.join(self.cterm.primary, "flux")).key
-                secondaryFluxKey = refSchema.find(refSchema.join(self.cterm.secondary, "flux")).key
-                primaryFluxErrKey = refSchema.find(refSchema.join(self.cterm.primary, "fluxErr")).key
-                secondaryFluxErrKey = refSchema.find(refSchema.join(self.cterm.secondary, "fluxErr")).key
-                refFlux1 = numpy.array([m[0].get(primaryFluxKey) for m in matches])
-                refFlux2 = numpy.array([m[0].get(secondaryFluxKey) for m in matches])
-                refFluxErr1 = numpy.array([m[0].get(primaryFluxErrKey) for m in matches])
-                refFluxErr2 = numpy.array([m[0].get(secondaryFluxErrKey) for m in matches])
-                refMag1 = -2.5*numpy.log10(refFlux1)
-                refMag2 = -2.5*numpy.log10(refFlux2)
-                refMag = self.cterm.transformMags(refMag1, refMag2)
-                refFlux = numpy.power(10.0, -0.4*refMag)
-                refFluxErr = self.cterm.propagateFluxErrors(refFluxErr1, refFluxErr2)
+
+                # extract the matched refCat as a Catalog for the colorterm code
+                refCat = afwTable.SimpleCatalog(matches[0].first.schema)
+                refCat.reserve(len(matches))
+                for x in matches:
+                    record = refCat.addNew()
+                    record.assign(x.first)
+
+                refMag, refMagErr = self.cterm.getCorrectedMagnitudes(refCat,
+                                                                      afwImage.Filter(calexp_md).getName())
+                # NOTE: mosaic assumes fluxes are in Jy
+                refFlux = (refMag*astropy.units.ABmag).to_value(astropy.units.Jy)
```

Comment:
This is a weird mix of astropy to compute the refFlux and the stack to compute refFluxErr. 
Does `afwImage.fluxFromABMag` work?

Reply:
I'm trying to get rid of all uses of that in the stack prior to deprecation. See DM-16903.

Unfortunately, astropy doesn't currently have a way to handle the error calculations directly (`astropy.uncertainty` isn't finished yet), otherwise I would have replaced the other one too.